packages/validators/src/validators/validateExperienceFields.ts

```diff
@@ -1,28 +1,44 @@
 import { type SchemaVersions } from '../types';
 import { ValidatorReturnValue } from './ValidatorReturnValue';
-import { Schema_2023_09_28 } from '../schemas';
-import { zodToContentfulError, CodeNames } from '../utils/zodToContentfulError';
+import { ExperienceSchema_2023_09_28 } from '../schemas';
+import { zodToContentfulError, CodeNames } from '@/utils/zodToContentfulError';
+import { validatePatternFields } from '@/validators/validatePatternFields';
 
 const VERSION_SCHEMAS = {
-  '2023-09-28': Schema_2023_09_28,
+  '2023-09-28': ExperienceSchema_2023_09_28,
 };
 
+// TODO: fix typing when the Entry type is exposed
+function isPattern(experience: any): boolean {
+  return experience.fields.componentSettings !== undefined;
+}
+
 /**
  *
  * @param experience The experience entry to validate
- * @param schemaVersionOverride Optional override for the schema version to validate against. By default the schema version is read from the experience entry
+ * @param schemaVersionOverride Optional override for the schema version to validate against.
+ * By default, the schema version is read from the experience entry
  * @returns object with success property and optional errors array
  */
 export const validateExperienceFields = (
+  // TODO: type this as Entry when the type is exposed
   experience: any,
```

Comment:
```suggestion
  experience: Entry,
```
Is there a reason not using `Entry` from the core package right away in here?

Reply:
Mmmh, do you know how we can type this actually correct for the CMA shape with the locales inside?
IIRC `Entry` itself types the CDA shape (already resolved for one locale) but maybe we can still get the CMA shape by use the `'WITH_ALL_LOCALES'` modifier which is a generic type of `Entry`.

Reply:
Judging by its usage in `user_interface` ([search results](https://github.com/search?type=code&auto_enroll=true&q=org%3Acontentful%20validateExperienceFields)), the correct type for the CMA use case would be `EntryProps` from `contentful-management/types` but I guess we don't want to add this dependency to the mono repo just for this single type.

Reply:
`contentful-management` was already indirectly installed. This also makes the `Entry` re-export redundant